drivers/sensor/max17055/max17055.c

```diff
@@ -251,26 +253,27 @@ static int max17055_sample_fetch(const struct device *dev,
 	return 0;
 }
 
-static int max17055_exit_hibernate(struct max17055_data *priv)
+static int max17055_exit_hibernate(const struct device *dev)
 {
 	LOG_DBG("Exit hibernate");
 
-	if (max17055_reg_write(priv, SOFT_WAKEUP, SOFT_WAKEUP_WAKEUP)) {
+	if (max17055_reg_write(dev, SOFT_WAKEUP, SOFT_WAKEUP_WAKEUP)) {
 		return -EIO;
 	}
-	if (max17055_reg_write(priv, HIB_CFG, HIB_CFG_CLEAR)) {
+	if (max17055_reg_write(dev, HIB_CFG, HIB_CFG_CLEAR)) {
 		return -EIO;
 	}
-	if (max17055_reg_write(priv, SOFT_WAKEUP, SOFT_WAKEUP_CLEAR)) {
+	if (max17055_reg_write(dev, SOFT_WAKEUP, SOFT_WAKEUP_CLEAR)) {
 		return -EIO;
 	}
 
 	return 0;
 }
 
-static int max17055_write_config(struct max17055_data *priv,
-				 const struct max17055_config *const config)
+static int max17055_write_config(const struct device *dev)
 {
+	const struct max17055_config *config = dev->config;
```

Comment:
where is config used here?

Reply:
Okey that link wasn't very helpful... It's used from line 277 through 282.